realm/realm-library/src/objectServer/java/io/realm/RealmApp.java

```diff
@@ -49,30 +57,7 @@
     // we might want to lift in the future. So any implementation details so ideally be made
     // with that in mind, i.e. keep static state to minimum.
 
-    // Default session error handler that just output errors to LogCat
-    private static final SyncSession.ErrorHandler SESSION_NO_OP_ERROR_HANDLER = new SyncSession.ErrorHandler() {
-        @Override
-        public void onError(SyncSession session, ObjectServerError error) {
-            if (error.getErrorCode() == ErrorCode.CLIENT_RESET) {
-                RealmLog.error("Client Reset required for: " + session.getConfiguration().getServerUrl());
-                return;
-            }
-
-            String errorMsg = String.format(Locale.US, "Session Error[%s]: %s",
-                    session.getConfiguration().getServerUrl(),
-                    error.toString());
-            switch (error.getErrorCode().getCategory()) {
-                case FATAL:
-                    RealmLog.error(errorMsg);
-                    break;
-                case RECOVERABLE:
-                    RealmLog.info(errorMsg);
-                    break;
-                default:
-                    throw new IllegalArgumentException("Unsupported error category: " + error.getErrorCode().getCategory());
-            }
-        }
-    };
+    private volatile static String CREATED_APP_ID = null;
```

Comment:
Just a generic thought about this:
I acknowledge the comment written just above this line but using static variables to controlling instantiation logic is kind of ugly. I don't think the class itself should be responsible for knowing whether it has been instantiated or not. Would it not make sense to hide it behind something else?

Reply:
Yes, this is super ugly and a hack. But it is one that is going to be removed shortly so this was the smallest amount of code needed to check it. Note, I improved the check and description around these variables. Hopefully, that makes it acceptable as a temp solution.